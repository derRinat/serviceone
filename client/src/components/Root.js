import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';

import Routers from './Routers';

const Root = ({ store, history }) => (

    <Provider store={ store }>
        <div>
            <Routers history={ history } store={store}/>
        </div>
    </Provider>

);

Root.propTypes = {
    store: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired
};

export default Root;
