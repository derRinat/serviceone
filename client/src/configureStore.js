import { createStore, applyMiddleware } from 'redux';
import { routerMiddleware, syncHistoryWithStore } from 'react-router-redux';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const configureStore = (history, initialState = {}) => {
    const middlewares = [
        thunk,
        routerMiddleware(history)
    ];

    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(...middlewares)
    );

    syncHistoryWithStore(history, store);

    return store;
};

export default configureStore;
