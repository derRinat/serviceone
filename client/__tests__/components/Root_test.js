
import React from 'react';
import configureStore from 'redux-mock-store';
import { createBrowserHistory } from 'history'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Root from '../../src/components/Root';

describe('Component: Root', () => {

    const properties = {
        store: configureStore()({}),
        history: createBrowserHistory()
    }

    it('renders without crashing', () => {
        const component = shallow(
            <Root { ...properties }/>
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
