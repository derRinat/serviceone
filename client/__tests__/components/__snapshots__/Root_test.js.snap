// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Root renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <Routers
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    />
    <Connect(Alerts) />
  </div>
</Provider>
`;
